pages/api/submit.ts

```diff
@@ -145,72 +145,74 @@ const processFormData = async (
+          if (fileItem.name) {
+            logMessage.info(`uploading: ${_key} - filename ${fileItem.name} `);
+            const { isValid, key } = await pushFileToS3(fileItem);
             if (isValid) {
-              uploadedFilesKeyUrlMapping.set(fileOrArray.name, key);
+              uploadedFilesKeyUrlMapping.set(fileItem.name, key);
               const splitKey = _key.split("-");
               if (splitKey.length > 1) {
                 const currentValue = fields[splitKey[0]] as Record<string, unknown>[];
-                currentValue[Number(splitKey[1])][splitKey[2]] = key;
+                currentValue[Number(splitKey[1])][`${splitKey[2]}-${index}`] = key;
               } else {
-                fields[_key] = key;
+                fields[`${_key}-${index}`] = key;
               }
             }
           }
-        } else if (Array.isArray(fileOrArray)) {
-          // An array will be returned in a field that includes multiple files
-          fileOrArray.forEach(async (fileItem, index) => {
-            if (fileItem.name) {
-              logMessage.info(`uploading: ${_key} - filename ${fileItem.name} `);
-              const { isValid, key } = await pushFileToS3(fileItem);
-              if (isValid) {
-                uploadedFilesKeyUrlMapping.set(fileItem.name, key);
-                const splitKey = _key.split("-");
-                if (splitKey.length > 1) {
-                  const currentValue = fields[splitKey[0]] as Record<string, unknown>[];
-                  currentValue[Number(splitKey[1])][`${splitKey[2]}-${index}`] = key;
-                } else {
-                  fields[`${_key}-${index}`] = key;
-                }
-              }
-            }
-          });
-        }
-      }
-      try {
-        await callLambda(
-          form.formID,
-          fields,
-          // pass in the language from the header content language... assume english as the default
-          req.headers?.["content-language"] ? req.headers["content-language"] : "en"
-        );
-
-        if (notifyPreview) {
-          const notifyPreviewResponse = await previewNotify(form, fields);
-          return res.status(201).json({ received: true, htmlEmail: notifyPreviewResponse });
-        } else {
-          return res.status(201).json({ received: true });
-        }
-      } catch (err) {
-        logMessage.error(err);
-        return res.status(500).json({ received: false });
+        });
       }
     }
-    // Local development and Heroku
-    else if (notifyPreview) {
-      return await previewNotify(form, fields).then((response) => {
-        return res.status(201).json({ received: true, htmlEmail: response });
-      });
+    try {
+      await callLambda(
+        form.formID,
+        fields,
+        // pass in the language from the header content language... assume english as the default
+        req.headers?.["content-language"] ? req.headers["content-language"] : "en"
+      );
+
+      if (notifyPreview) {
```

Comment:
This logic will never be triggered because of the previous logic check against `notifyPreview` on line 154.  This logic block can be safely removed.  We'll eventually pull the complete 'notifyPreview' feature out of the app as it's not really used.